src/richie/apps/core/fields/duration.py

```diff
@@ -0,0 +1,275 @@
+                except ValueError:
+                    raise exceptions.ValidationError(
+                        _(
+                            "A composite duration should be a round number of intervals of time."
+                        ),
+                        code="invalid_duration",
+                    )
+
+                # Check that the duration is positive
+                if duration <= 0:
+                    raise exceptions.ValidationError(
+                        _("A composite duration should be positive."),
+                        code="negative_duration",
+                    )
+
+            choices = list(dict(self.intervals))
+
+            # Search the interval in this set
+            if interval not in choices:
+                raise exceptions.ValidationError(
+                    self.error_message,
+                    code="invalid_interface",
+                    params={"value": interval},
+                )
+
+        if value is None and not self.null:
+            raise exceptions.ValidationError(self.error_messages["null"], code="null")
+
+        if not self.blank and not value:
+            raise exceptions.ValidationError(self.error_messages["blank"], code="blank")
+
+    def formfield(self, **kwargs):
+        """Declare the above CompositeDurationFormField as form field for this model field."""
+        kwargs = {
+            **kwargs,
+            "form_class": CompositeDurationFormField,
+            "widget": CompositeDurationWidget(
+                choices=tuple(
+                    (name, strings[0]) for name, strings in self.intervals.items()
+                ),
+                default_interval=self.default_interval,
+            ),
+        }
+        return super().formfield(**kwargs)
+
+    def contribute_to_class(self, cls, name, private_only=False):
+        """
+        Add a method to the model using this field to enable the common `get_FIELD_display`
+        pattern to display the human readable version of a field based on choices.
+
+        We choose to display a sentence of the form: "5 days".
+        """
+        super().contribute_to_class(cls, name, private_only=private_only)
+        choices = self.intervals
+
+        def func(self, *_args):
+            """
+            we add `*args` because this method is used in a "render_model" template tag
+            and it passes a "request" argument when calling the method.
+            For more information, see:
+            http://docs.django-cms.org/en/latest/how_to/frontend_models.html
+            """
+            value = getattr(self, name)
+            if value:
+                duration, interval = value
+                duration = int(duration)
+                count_index = 1 if duration > 1 else 0
+                return "{duration:d} {interval!s}".format(
+                    duration=duration, interval=choices[interval][count_index]
+                )
+            return ""
```

Comment:
Can you elaborate what your suggestion is?

Reply:
https://williamdurand.fr/2013/06/03/object-calisthenics/#2-dont-use-the-else-keyword

Reply:
The idea is to minimize the indentation with an early return:

```python
        def func(self, *_args):
            """
            we add `*args` because this method is used in a "render_model" template tag
            and it passes a "request" argument when calling the method.
            For more information, see:
            http://docs.django-cms.org/en/latest/how_to/frontend_models.html
            """
            value = getattr(self, name)
            if not value:
                return ""

            duration, interval = value
            duration = int(duration)
            count_index = 1 if duration > 1 else 0
            return "{duration:d} {interval!s}".format(
                duration=duration, interval=choices[interval][count_index]
            )
```